app/code/Magento/Quote/Model/QuoteManagement.php

```diff
@@ -355,6 +356,13 @@ public function placeOrder($cartId, PaymentInterface $paymentMethod = null)
         if ($quote->getCheckoutMethod() === self::METHOD_GUEST) {
             $quote->setCustomerId(null);
             $quote->setCustomerEmail($quote->getBillingAddress()->getEmail());
+            if ($quote->getCustomerFirstname() === null && $quote->getCustomerLastname() === null) {
+                $quote->setCustomerFirstname($quote->getBillingAddress()->getFirstname());
+                $quote->setCustomerLastname($quote->getBillingAddress()->getLastname());
+                if ($quote->getBillingAddress()->getMiddlename() === null) {
```

Comment:
Is this check by design, or did you mean to check `$quote->hasCustomerMiddlename` here?

Reply:
Yes in original Pr i was check if `$quote->hasCustomerMiddlename` but now i don't know how i did this, i'm will add a propper check